### PR TITLE
Replaces watcher freeze blasts with a new mechanic

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
tweak: The watcher now no longer fires freezing bolts, instead causing petrification to those that look at it for too long.
/:cl:

I've been playing a lot of shaft miner lately, and it's become very clear to me that watchers are poorly-designed. Whereas goliaths can be beaten with timing and legions with rushing,, the thing determining the watcher's difficulty is how buggy the shots are. The only time they ever consistently hit or miss is when you're parallel to them; they otherwise hit or miss seemingly at random if you're within two tiles or so.

I replaced their firing with a new mechanic. Now, instead of firing a freeze-bolt every 3 seconds, they will induce petrification in their target every 7.5 seconds.

Petrification is a status effect that lasts for 5 seconds total. Every fifth of a second, it will tick; if the target is looking directly at the watcher, they will gain 20% petrification, and otherwise lose 1% petrification. At 100% petrification, the target will be stunned for 10 seconds; otherwise, the effect will fade harmlessly.

Each mob on lavaland represents to me a test; goliaths test your timing, and legions test your speed and damage output. Watchers were meant to test dodging, but the bugginess of their projectiles and aiming meant that they fail in this. With this mechanic, they test how creative you can get with spacing and how to orient yourself in relation to the watcher.

Playtested with the crusher and KA; they're more fun to fight in both cases, in my opinion. Not having the constant slowing hits means that it's much less infuriating to both fight and flee from watchers.

### CODE REVIEW

**Comment by Dorsisdwarf** [comment]
So you have to orient yourself away from them or get stoned in 1 second?

Also a 10 second stun is basically death if theres anything other than a watcher around

File: code/datums/status_effects/debuffs.dm [hunk 1]
```diff
@@ -434,3 +434,47 @@
...
+	tick_interval = 2
+	alert_type = /obj/screen/alert/status_effect/petrification
+	var/petrification_percentage = 0 //How petrified are we? At 100%, we'll be stunned for 10 seconds.
+	var/mob/living/petrifier
+
+/datum/status_effect/petrification/tick()
+	if(QDELETED(petrifier) || petrifier.stat)
+		qdel(src)
+		return
+	if(get_dist(owner, petrifier) < 7 && owner.dir == get_dir(owner, petrifier)) //we're looking at it
```
Comment (RemieRichards) [inline code/datums/status_effects/debuffs.dm]: this doesn't count walls or anything properly.

File: code/datums/status_effects/debuffs.dm [hunk 2]
```diff
@@ -434,3 +434,47 @@
...
+		playsound(owner, 'sound/magic/fleshtostone.ogg', 50, TRUE, frequency = 2)
+		petrification_percentage += 20
+	else
+		petrification_percentage = max(0, petrification_percentage - 1)
+	if(petrification_percentage >= 100)
+		owner.visible_message("<span class='warning'>[owner]'s body locks up!</span>", "<span class='userdanger'>Your limbs lock up! You can't move!</span>")
+		playsound(owner, 'sound/effects/break_stone.ogg', 75, TRUE, frequency = 0.5)
+		owner.Stun(100)
+		qdel(src)
+		return
```
Comment (RemieRichards) [inline code/datums/status_effects/debuffs.dm]: you know what's wrong here.

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/basilisk.dm [hunk 3]
```diff
@@ -56,6 +46,24 @@
...
+		return
+	var/mob/living/L = target
+	if(L.has_status_effect(STATUS_EFFECT_PETRIFICATION) || L.incapacitated())
+		return
+	visible_message("<span class='warning'>[src] [ranged_message] [target]</span>")
+	to_chat(L, "<span class='boldannounce'>Your body starts to lock up as you look at [src]!</span>")
+	if(gaze_sound)
+		playsound(src, gaze_sound, 75, FALSE)
+	L.apply_status_effect(STATUS_EFFECT_PETRIFICATION)
+	var/datum/status_effect/petrification/P = L.has_status_effect(STATUS_EFFECT_PETRIFICATION)
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/basilisk.dm]: ```apply_status_effect``` returns the effect, there's no need for this line.

**Comment by KorPhaeron** [comment]
@Ergovisavi

**Comment by banthebantz** [comment]
how about the watcher only petrifies anything in a line directly in front of it (so actually in it's stare sprite wise), compensate by increasing speed of petrification

**Comment by MimicFaux** [comment]
My concern is if this stacks; attacking a watcher tendril is already tricky enough as it is.
If 2/3 watchers are all contributing to the petrification it would be near suicidal to attempt.

Side note: Does the status effect use the same overlay like the wizard spell does?

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/basilisk.dm [hunk 4]
```diff
@@ -56,6 +46,24 @@
...
+	if(!isliving(target))
+		return
+	var/mob/living/L = target
+	if(L.has_status_effect(STATUS_EFFECT_PETRIFICATION) || L.incapacitated())
+		return
+	visible_message("<span class='warning'>[src] [ranged_message] [target]</span>")
+	to_chat(L, "<span class='boldannounce'>Your body starts to lock up as you look at [src]!</span>")
+	if(gaze_sound)
+		playsound(src, gaze_sound, 75, FALSE)
+	L.apply_status_effect(STATUS_EFFECT_PETRIFICATION)
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/basilisk.dm]: this should just be `L.apply_status_effect(STATUS_EFFECT_PETRIFICATION, src)` with the on_creation properly setting the petrifier it gets as an argument.

**Comment by Ergovisavi** [comment]
So you "fixed" the watcher's simple and readable mechanic by adding something obtuse and hard to understand?

I need to actually see this in action before making any judgements but by default I strongly dislike the implications of this.

Basilisks/Watchers exist as sentries and force multipliers for other mobs, getting near them is intended to be a dangerous experience and if this just boils down looking away from the mob for a second to completely ignore it, no thanks.

**Comment by Xhuis** [comment]
The role of watchers is warped by how inconsistent their attacks are with bugginess. Their role as enforcers is either completely subverted by shots missing when they shouldn't, or gigantically blown out of proportion when they hit from several tiles away.

They still do serve the role with petrification, in that it makes taking on other groups much harder as you have to watch there you look. Even though it may much different, it's by far less unenjoyable and frustrating to fight against when you aren't being slowed constantly by hits you can't avoid.

**Comment by Repukan** [comment]
It is very frustrating to fight against something that will always hit with no way to dodge, even more so than being hit by a freezing bolt from far away.

**Comment by Ergovisavi** [comment]
Basilisks/Watchers are not supposed to be "fun to fight". They exist to make you consider your actions, your current health and healing supplies, and make a judgement call on whether or not agressing for the diamonds or taking another route is the best course of action. Do I have a legion core on me? Is there anything else nearby? Forcing you to stop and think is the intended goal.

Lavaland had this effect on them where they became significantly weaker due to their environment, IE: You can actually survive in your surroundings and you have station movement. Previously you had neither and so they were a genuinely scary thing to get near.

Given the overall differences between their original environment and the current one, I'm fine with a change to their behaviour but this one just seems hard to read for no benefit.

**Comment by Xhuis** [comment]
I'm open to any suggestions. I have no particular attachment to petrification other than that it replaces the bad way that watchers currently operate.

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/basilisk.dm (hunk 4 above)
Comment (Xhuis) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/basilisk.dm]: I'll remember one of these days.

**Comment by Ratvar** [comment]
Remove petrification if you attack watcher from buttsex side in melee aka it looses concentration(add prototype axe kinetic thing for wombo combo)? Remove all petri stacks if you kill one? Half petrification for nearsightned and people in welding glasses, et cetera? Fight watcher hive with welding mask on.  Beware of ashliznurds throwning spears at you from screen border. You can print those in cargo autolathes, right? Convert dead corpses near watchers into TurnToStone statues, as alternative to legions and megafauna gibbing, so you can break em open to rethrive corpses with all loot if you want or use for advanced greff as tatortoe xenobiologist? Add "Second Eye Of God" miniboss thing? Make them guard statues and add one new max watcher to tendrill, so you can cultivate them up?

**Comment by KorPhaeron** [comment]
The mining man says no and I trust him with my life

**Comment by Xhuis** [comment]
I would trust him with my life too fam, don't worry

**Pull request closed without merging**